<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>898</width>
    <height>801</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>David &amp; the Johnsons</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="canvas" native="true">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>0</y>
      <width>531</width>
      <height>571</height>
     </rect>
    </property>
    <widget class="EditorFrame" name="frame" native="true">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>512</width>
       <height>512</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="subFrame">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>530</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>25</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
    </widget>
    <widget class="QPushButton" name="moveLeft">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>530</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>25</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>&lt;</string>
     </property>
    </widget>
    <widget class="QPushButton" name="addFrame">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>530</y>
       <width>30</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>25</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>+</string>
     </property>
    </widget>
    <widget class="QPushButton" name="moveRight">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>530</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>25</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="currFrameLabel">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>540</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Current Frame: 1</string>
     </property>
    </widget>
    <widget class="QLabel" name="totalFrameLabel">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>540</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Total Frames: 1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="duplicateFrame">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>530</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Duplicate Frame</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="toolsWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>230</y>
      <width>361</width>
      <height>561</height>
     </rect>
    </property>
    <widget class="QPushButton" name="sizeButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Frame Size</string>
     </property>
    </widget>
    <widget class="QPushButton" name="eraseButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>100</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Erase</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="drawButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Draw</string>
     </property>
    </widget>
    <widget class="QPushButton" name="fileButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>190</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>File</string>
     </property>
    </widget>
    <widget class="QPushButton" name="symmetricalButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>390</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Symmetry</string>
     </property>
    </widget>
    <widget class="QPushButton" name="colorButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>40</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Color</string>
     </property>
    </widget>
    <widget class="QPushButton" name="shiftButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>390</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Shift</string>
     </property>
    </widget>
    <widget class="QPushButton" name="undoButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>480</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Undo</string>
     </property>
    </widget>
    <widget class="QPushButton" name="clearButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>290</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QLabel" name="frameSizeDisplay">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Frame Size = 16</string>
     </property>
    </widget>
    <widget class="QLabel" name="penSizeDisplay">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>40</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Pen Size = 1x1</string>
     </property>
    </widget>
    <widget class="QLabel" name="colorDisplayLabel">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>40</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Color =</string>
     </property>
    </widget>
    <widget class="QPushButton" name="colorDisplay">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="redoButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>480</y>
       <width>150</width>
       <height>75</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>45</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Redo</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="previewView" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>351</width>
      <height>251</height>
     </rect>
    </property>
    <widget class="QPushButton" name="pauseButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>205</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Play/Pause</string>
     </property>
    </widget>
    <widget class="QSlider" name="fpsSlider">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>210</y>
       <width>211</width>
       <height>31</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="previewImage">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>10</y>
       <width>190</width>
       <height>190</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QStackedWidget" name="optionsBox">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>590</y>
      <width>531</width>
      <height>211</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <widget class="QWidget" name="penView">
     <widget class="QPushButton" name="penSize1">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>32</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>1</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="penSize2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>32</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>2</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="penSize3">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>32</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>3</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="penSize4">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>32</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>4</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>10</y>
        <width>111</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
        <bold>true</bold>
        <kerning>false</kerning>
       </font>
      </property>
      <property name="text">
       <string>Brush Size</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="colorView"/>
    <widget class="QWidget" name="symmetryView">
     <widget class="QPushButton" name="HorizontalButton">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>50</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>30</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Horizontal</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="VerticalButton">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>50</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>30</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Vertical</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="fileView">
     <widget class="QPushButton" name="LoadButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>40</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>50</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Load</string>
      </property>
     </widget>
     <widget class="QPushButton" name="SaveBu">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>40</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>45</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="sizeView">
     <widget class="QPushButton" name="size8Button">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>64</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>8</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="size16Button">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>64</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>16</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="size64Button">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>64</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>64</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="size32Button">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>60</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>64</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>32</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>10</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>24</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Frame Size</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="shiftView">
     <widget class="QPushButton" name="shiftLeftButton">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>50</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="text">
       <string>Left</string>
      </property>
     </widget>
     <widget class="QPushButton" name="shiftRightButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="text">
       <string>Right</string>
      </property>
     </widget>
     <widget class="QPushButton" name="shiftUpButton">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>50</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="text">
       <string>Up</string>
      </property>
     </widget>
     <widget class="QPushButton" name="shiftDownButton">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>50</y>
        <width>101</width>
        <height>111</height>
       </rect>
      </property>
      <property name="text">
       <string>Down</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EditorFrame</class>
   <extends>QWidget</extends>
   <header>editorframe.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
